app/server/controllers/login.ts

```diff
@@ -25,54 +45,74 @@ function getLogout(req: Request, res: Response) {
   });
 }
 
-function postLogin(getOnlineHearing) {
-  return async(req: Request, res: Response, next: NextFunction) => {
-    const email = req.body['email-address'];
-    const validationMessage = loginEmailAddressValidation(email);
-    if (validationMessage) {
-      const emailAddress = {
-        value: email,
-        error: validationMessage
-      };
-      return res.render('login.html', { emailAddress });
+function getIdamCallback(
+  getToken: (code: string, protocol: string, hostname: string) => TokenResponse,
+  getUserDetails: (accessToken: string) => UserDetails,
+  getOnlineHearing: (email:string) => superAgent.Response,
+  getRedirectUrl: (protocol: string, hostname: string) => string) {
+  return async (req: Request, res: Response, next: NextFunction) => {
+
+    const code: string = req.query.code;
+
+    console.log("Code: " + code);
+
+    if (!code) {
+      return getLogin(getRedirectUrl, '/login')(req, res);
```

Comment:
Would this not end up in a redirect loop? login -> IDAM -> callback (no code) -> login -> IDAM -> callback (no code) etc.

If there isn't a code should we consider this as a server error and show the user the 500 page?

Reply:
No this sends them to the idam login page. It happens when the user has just registered or changed their password. We then need to send them to the login page.

Reply:
Ok, my misunderstanding, cheers